ocs_ci/ocs/resources/mcg.py

```diff
@@ -114,6 +116,28 @@ def s3_verify_bucket_exists(self, bucket):
             logger.info(f"{bucket.name} does not exist")
             return False
 
+    def verify_s3_object_integrity(self, original_object, result_object, awscli_pod):
+        """
+        Verifies checksum between orignial object and result object on an awscli pod
+        Args:
+            orignal_object(str) : The Object that is uploaded to the s3 bucket
+            result_object(str) :  The Object that is downloaded from the s3 bucket
+
+        Returns:
+              bool: True if checksum matches, False otherwise
+
+        """
+        md5sum = awscli_pod.exec_cmd_on_pod(
+            command=f'md5sum {original_object} {result_object}'
+        )
+        md5sum_original, md5sum_result = md5sum.split()[0], md5sum.split()[2]
+        if md5sum_original == md5sum_result:
+            logger.info(f'Passed: MD5 comparison for {original_object} and {result_object}')
+            return True
+        else:
+            logger.info('Data Corruption Found')
```

Comment:
logger.error

Reply:
Would suggest adding more information to the log - 
`logger.error(f'Failed MD5 comparison of {original_object} and {result_object} - {md5sum_original} ≠ {md5sum_result})`

Reply:
ack